### DISCUSSION TITLE
Allow multi-return fn to be accepted as fn parameters

### DISCUSSION THREAD

**Discussion by depperm** [vlang/v#14669] [Ideas]
if `fn a() (int,int)` and `fn b(int,int)` I should be able to do `b(a())`. Currently I get `error: expected 2 arguments, but got 1`

This seems to conflict with doing `q := a()` which gives error `error: assignment mismatch: 1 variable(s) but 'a()' returns 2 value(s)`. If V knows 2 values are returned, I should be able to pass them as arguments

**Comment by dumblob** [comment]
This is closely related to the evergreen issue with keyword arguments: https://github.com/vlang/v/pull/11135#issuecomment-924823890 .

**Comment by JalonSolov** [comment]
I don't see how keyword arguments relates to this at all.  This one is simply about V not paying attention to the number of arguments returned vs listed.

**Comment by dumblob** [comment]
@JalonSolov do you mean *V* shall not support this requested functionality for functions (a) accepting keyword arguments or (b) both positional **and** keyword args?

That would introduce strong inconsistency in *V* and would be clearly a missing functionality.

So if *V* will want to implement the requested functionality, then this has to work also for functions accepting both positional and keyword arguments. In which case it is highly probable that this requested functionality would be implemented as part of "true keywoard arguments" PR. Therefore the "close relation" as I called it.

Btw. surprisingly `println( a() )` works - another weird thing about `println()` which shall be IMHO fixed before *V 1.0*.

**Comment by JalonSolov** [comment]
I made no claims about what V shall or shall not support.  I just said that the problem with _this specific issue_ is that V is not checking for how many things `a()` returns, it is only looking at the fact that there is a single `a()` as parameters for `b()`.  It _should_ work, but it doesn't.

That has _nothing_ to do with support of keyword arguments.
